api/image_pipeline/pipeline_image_util.py

```diff
@@ -0,0 +1,671 @@
+    """
+    url = f"https://errata.devel.redhat.com/api/v1/variants/{variant_id}"
+    response = request_with_kerberos(url)
+
+    try:
+        return response.json()['data']['attributes']['relationships']['product_version']['id']
+    except Exception:
+        raise exceptions.ProductIdNotFound(f"Product ID not found for variant `{variant_id}`")
+
+
+def cdn_to_github(cdn_name: str, version: str) -> tuple[Github, Distgit, Brew]:
+    """
+    Driver function for the CDN -> GitHub pipeline
+
+    :cdn_name: The name of the CDN repo
+    :version: The OCP version
+    """
+
+    # Brew
+    brew_name = cdn_to_brew(cdn_name)
+    brew_id = get_brew_id(brew_name)
+
+    github_object, distgit_object, brew_object = brew_to_github(brew_name, version)
+
+    brew_object.brew_id = brew_id
+    brew_object.brew_build_url = f"https://brewweb.engineering.redhat.com/brew/packageinfo?packageID={brew_id}"
+    brew_object.brew_package_name = brew_name
+
+    return github_object, distgit_object, brew_object
+
+
+def get_cdn_payload(cdn_repo_name: str, variant: str) -> CDN:
+    """
+    Function to get the CDN payload for slack (since it's needed in multiple places.
+
+    :cdn_repo_name: The name of the CDN repo
+    :variant: The 8Base-RHOSE variant
+    """
+    cdn_repo_id = get_cdn_repo_id(cdn_repo_name)
+    variant_id = get_variant_id(cdn_repo_name, variant)
+    product_id = get_product_id(variant_id)
+
+    cdn_object = CDN()
+    cdn_object.cdn_repo_id = cdn_repo_id
+    cdn_object.cdn_repo_name = cdn_repo_name
+    cdn_object.cdn_repo_url = f"https://errata.devel.redhat.com/product_versions/{product_id}/cdn_repos/{cdn_repo_id}"
+    cdn_object.variant_name = variant
+    cdn_object.variant_id = variant_id
+
+    return cdn_object
+
+
+def cdn_to_delivery_payload(cdn_repo_name: str):
+    """
+    Function to get the CDN payload for slack (since it's needed in multiple places.
+
+    :cdn_repo_name: The name of the CDN repo
+    """
+    delivery_repo_name = cdn_to_delivery(cdn_repo_name)
+    delivery_repo_id = get_delivery_repo_id(delivery_repo_name)
+
+    delivery_object = Delivery()
+    delivery_object.delivery_repo_id = delivery_repo_id
+    delivery_object.delivery_repo_name = delivery_repo_name
+    delivery_object.delivery_repo_url = f"https://comet.engineering.redhat.com/containers/repositories/{delivery_repo_id}"
+
+    return delivery_object
+
+
+# Delivery stuff
+def delivery_repo_is_available(name: str) -> bool:
+    """
+    Function to check if the given delivery repo exists
+
+    :name: Name of the delivery repo
+    """
+    try:
+        _ = get_delivery_repo_id(name)
+        return True
+    except exceptions.DeliveryRepoIDNotFound:
+        return False
+
+
+def brew_from_delivery(delivery_repo: str) -> str:
+    """
+    Function to get the brew name from the delivery repo
+    """
+    url = f"https://pyxis.engineering.redhat.com/v1/repositories/registry/registry.access.redhat.com/repository/{delivery_repo}/images"
```

Comment:
"https://pyxis.engineering.redhat.com/v1/repositories/registry/registry.access.redhat.com/repository/openshift4/ose-csi-driver-nfs-rhel8/images?page_size=1&include=data.brew" This would be much faster

Reply:
Cool, thanks for the info. Will convert to an issue and will work a separate refinement PR.